eppoclient/configurationrequestor_test.go

```diff
@@ -0,0 +1,106 @@
+package eppoclient
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockHttpClient struct {
+	mock.Mock
+}
+
+func (m *mockHttpClient) get(url string) (string, error) {
+	args := m.Called(url)
+	return args.String(0), args.Error(1)
+}
+
+func Test_FetchAndStoreConfigurations_DoesNotDeserializeTwice(t *testing.T) {
+	mockHttpClient := new(mockHttpClient)
+	mockConfigStore := newConfigurationStore()
+	configRequestor := newConfigurationRequestor(mockHttpClient, mockConfigStore)
+
+	// Mock the HTTP client to return a fixed response
+	mockResponse1 := `
+	{
+		"createdAt": "2024-04-17T19:40:53.716Z",
+		"environment": {
+			"name": "Test"
+		},
+		"flags": {
+			"empty_flag": {
+				"key": "empty_flag",
+				"enabled": true,
+				"variationType": "STRING",
+				"variations": {},
+				"allocations": [],
+				"totalShards": 10000
+			}
+		}
+	}
+	`
+	mockCall := mockHttpClient.On("get", UFC_ENDPOINT).Return(mockResponse1, nil)
+
+	// First fetch and store configurations
+	configRequestor.FetchAndStoreConfigurations()
+
+	// Store the current hash
+	firstHash := configRequestor.storedFlagConfigsHash
+
+	// Fetch and store configurations again
+	configRequestor.FetchAndStoreConfigurations()
+
+	// Store the hash after the second fetch
+	secondHash := configRequestor.storedFlagConfigsHash
+
+	// Assert that the hash has not changed
+	assert.Equal(t, firstHash, secondHash)
+
+	// Assert that the configurations were only deserialized once
```

Comment:
Sorry just realized that this is NOT asserted here - will augment the PR and ping you guys when ready. 🙇